src/components/Overlay/Overlay.scss

```diff
@@ -0,0 +1,45 @@
+@use '../variables';
+
+$block: '.#{variables.$ns}overlay';
+
+#{$block} {
+    position: absolute;
+    inset: 0;
+
+    display: flex;
+    visibility: hidden;
+    justify-content: center;
+    align-items: center;
+
+    opacity: 0;
+
+    transition:
+        visibility 0.1s,
+        opacity 0.1s linear;
+
+    &_visible {
+        visibility: visible;
+        opacity: 1;
+    }
+
+    &__background {
+        position: absolute;
+        inset: 0;
+
+        opacity: 0.8;
+
+        &_style {
+            &_base {
+                background-color: var(--g-color-base-background);
+            }
+
+            &_float {
+                background-color: var(--g-color-base-float);
+            }
+        }
+    }
+
+    &__children {
+        z-index: 0;
```

Comment:
That's because of how opacity works in html. When z-index is not set, container would be drawn on top of its children. For children with opacity, though, it would work even without z-index.

Reply:
In order to not depend on stacking context switching due to opacity change, you should set `isolation: isolate` on the root, and both z-index on background and children elements with values: 0 and 1, respectively. So children will be always on top of background.